pkgs/development/cuda-modules/setup-hooks/auto-fix-elf-files.sh

```diff
@@ -0,0 +1,62 @@
+# shellcheck shell=bash
+# List all dynamically linked ELF files in the outputs and apply a generic fix
+# action provided as a parameter (currently used to add the CUDA or the
+# cuda_compat driver to the runpath of binaries)
+echo "Sourcing cuda/fix-elf-files.sh"
+
+# Returns the exit code of patchelf --print-rpath.
+# A return code of 0 (success) means the ELF file has a dynamic section, while
+# a non-zero return code means the ELF file is statically linked (or is not an
+# ELF file).
+elfHasDynamicSection() {
+  local libPath
```

Comment:
`="$1"; shift 1` I guess

Reply:
Is there any particular reason to do that? Does it invalidate `$1` if we try to use it later in the body?

Reply:
It changes the meaning of/reassigns `$1`

Reply:
What is the motivation for shifting, if we don't touch arguments anymore?

Reply:
cargo culting on my part really, but I was thinking that with `shift` we'd always have only 1 way of referring to the value at any chosen time, instead of `$1` + `$libpath`

Reply:
Ok. I guess there's nothing wrong with doing that, I was just curious :slightly_smiling_face: